app/presenters/tutorial_view_step_collection_presenter.rb

```diff
@@ -0,0 +1,69 @@
+class StepCollectionPresenter
+  def initialize(step_object_arr)
+    @steps = step_object_arr
+    @counter = 0
+    @steps_info_arr = []
+    @main_steps_arr = []
+  end
+
+  def set_step(step)
+    @step = step
+  end
+  
+  def get_step
+    @step
+  end
+
+  def empty_check
+    if @steps.empty?
+      get_main_steps_arr()
+    else
+      get_steps_ordering_array()
+    end
+  end
+
+  def reset_steps_info_arr
+    @steps_info_arr = []
+  end
+
+  def get_main_steps_arr()
+    return @main_steps_arr
+  end
+
+  def get_start_time()
+      @step.start_time
+  end
+
+  def get_end_time()
+      @step.end_time
+  end
+
+  def get_description()
+    @step.description
+  end
+  
+  def content
+      "#{@step.title} - (#{@step.video_time_conversion_for_steps})"
+  end
+
+  def get_steps_ordering_array()
```

Comment:
Assignment Branch Condition size for get_steps_ordering_array is too high. [16.16/15]